### ISSUE TITLE
Add a `Download my solution` button to the Completion Modal

### ISSUE THREAD

**Issue by Bouncey** [freeCodeCamp/learn#107]
As we are not storing user challenge solutions in the database any more, we should offer a `Download my solution` button on the `<CompletionModal />`

This button will look something like 
```jsx
<Button
  block={true}
  bsSize='lg'
  bsStyle='primary'
  download={`${challenge.dashedName}.json`}
  href={
    `data:text/json;charset=utf-8,${
      encodeURIComponent(JSON.stringify(files))
    }`
  }
  >
  Download my solution
</Button>
```

The button will allow a user to download their solution in `JSON` format. 

@utsab @imcodingideas

**Comment by tchaffee** [comment]
@Bouncey I'm not clear on why it's important that we won't be storing use solutions in the database? Would there be a technical challenge to downloading all of your solutions? Are solutions stored locally? Hopefully? Maybe not. I just went back to a previously completed lesson and my code is not there. That's surprising. Is that intended? I would expect to be able to go back to stuff I completed and see my own solution.

**Comment by nikrb** [comment]
@Bouncey I have a solution for this [here](https://github.com/nikrb/learn/tree/enh/download-solution).
I pass the dashedName as a prop to CompletionModal but not confident that's the right way to do it.
```
src/templates/Challenges/classic/Show.js
src/templates/Challenges/components/CompletionModal.js
```

**Comment by Bouncey** [comment]
Closed via #191
